wrappers/golang/curves/bls12377/g2_msm.go

```diff
@@ -49,3 +49,33 @@ func G2Msm(scalars core.HostOrDeviceSlice, points core.HostOrDeviceSlice, cfg *c
 	err := (cr.CudaError)(__ret)
 	return err
 }
+
+func G2PrecomputeBases(points core.HostOrDeviceSlice, precompute_factor int32, _c int32, ctx *cr.DeviceContext, output_bases core.HostOrDeviceSlice) cr.CudaError {
+	core.PrecomputeBasesCheck(points, precompute_factor, output_bases)
+
+	var pointsPointer unsafe.Pointer
+	if points.IsOnDevice() {
+		pointsPointer = points.(core.DeviceSlice).AsPointer()
+	} else {
+		pointsPointer = unsafe.Pointer(&points.(core.HostSlice[G2Affine])[0])
+	}
+	cPoints := (*C.g2_affine_t)(pointsPointer)
+
+	cPointsLen := (C.int)(points.Len())
+	cPrecomputeFactor := (C.int)(precompute_factor)
+	c_C := (C.int)(_c)
+	cPointsIsOnDevice := (C._Bool)(points.IsOnDevice())
+	cCtx := (*C.DeviceContext)(unsafe.Pointer(&ctx))
```

Comment:
I think this might cause an invalid `cCtx` since `ctx` is already a pointer to a DeviceContext

```suggestion
	cCtx := (*C.DeviceContext)(unsafe.Pointer(ctx))
```